### PR TITLE
Update properties

### PR DESCRIPTION
**Pull request by l2280212**
Add **effectiveMode** to `iotdb-system.properties`. The **effectiveMode** parameter determines the mode in which changes to the configuration take effect. It has three options:

first_start: Changes are only allowed before the service is started for the first time.
restart: Changes will take effect after the service is restarted. 
hot_reload: Changes will take effect immediately without needing to restart the service.

### CODE REVIEW

File: iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties [hunk 1]
```diff
@@ -227,45 +247,54 @@ dn_seed_config_node=127.0.0.1:10710
...
+# effectiveMode: restart
 # Datatype: boolean
 # dn_rpc_thrift_compression_enable=false
 
-# if true, a snappy based compression method will be called before sending data by the network
-# Datatype: boolean
-# this feature is under development, set this as false before it is done.
-# dn_rpc_advanced_compression_enable=false
-
+# the number of rpc selector
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties:259]: dn_rpc_advanced_compression_enable is used yet?

File: iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties [hunk 2]
```diff
@@ -227,45 +247,54 @@ dn_seed_config_node=127.0.0.1:10710
...
 
+# weather enable thrift compression
+# effectiveMode: restart
 # Datatype: boolean
 # dn_rpc_thrift_compression_enable=false
 
-# if true, a snappy based compression method will be called before sending data by the network
-# Datatype: boolean
-# this feature is under development, set this as false before it is done.
-# dn_rpc_advanced_compression_enable=false
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties:239]: dn_rpc_advanced_compression_enable is used
image

**Review by OneSizeFitsQuorum — COMMENTED** [review]
Remove unused code too?

File: iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties [hunk 3]
```diff
@@ -1041,37 +1197,45 @@ data_replication_factor=1
 # 2. The candidate size of seq tsfile in cross space compaction will be smaller than target_compaction_file_size * 1.5.
 # In most cases, the target file size of cross compaction won't exceed this threshold, and if it does, it will not be much larger than it.
 # default is 2GB
+# effectiveMode: restart
```
[the remark this answers is not in the document]
Comment (shuwenwei) [reply iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties:1200]: max_inner_compaction_candidate_file_num
target_compaction_file_size
max_cross_compaction_candidate_file_num
max_cross_compaction_candidate_file_size
min_cross_compaction_unseq_file_level

File: iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties [hunk 4]
```diff
@@ -1082,54 +1246,65 @@ data_replication_factor=1
...
 
 # The interval of compaction task schedule
+# effectiveMode: restart
 # Datatype: long, Unit: ms
 # compaction_schedule_interval_in_ms=60000
 
 # The interval of compaction task submission
+# effectiveMode: restart
 # Datatype: long, Unit: ms
 # compaction_submission_interval_in_ms=60000
```
Comment (shuwenwei) [inline iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties:1267]: This configuration is no longer useful

Comment (l2280212) [reply iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties:1273]: fix it

File: iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties (hunk 3 above)
[the remark this answers is not in the document]
Comment (l2280212) [reply iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties]: fix it

File: iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties (hunk 1 above)
Comment (l2280212) [reply iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties:264]: fix it

**Review by shuwenwei — APPROVED** [review]

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**